### PR TITLE
[Merged by Bors] - chore: deprecate AlgEquiv.map_* lemmas

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
These are redundant.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Ruben-VandeVelde** [comment]
!bench

File: Mathlib/Algebra/Algebra/Equiv.lean [hunk 1]
```diff
@@ -196,18 +196,22 @@ theorem coe_ringEquiv_injective : Function.Injective ((↑) : (A₁ ≃ₐ[R] A
   fun _ _ h => ext <| RingEquiv.congr_fun h
 #align alg_equiv.coe_ring_equiv_injective AlgEquiv.coe_ringEquiv_injective
 
+@[deprecated map_add (since := "2024-06-20")]
```
Comment (grunweg) [inline Mathlib/Algebra/Algebra/Equiv.lean:199]: Should this be `_root_.map_add`? Reading just the code, I cannot tell why this shouldn't have a `_root_`, but e.g. `map_smul` should. Same for the next three items.

**Review by grunweg — COMMENTED** [review]
Seems plausible to me - I have one question.
With that resolved, feel free to maintainer merge on my behalf.

File: Mathlib/Algebra/Algebra/Equiv.lean (hunk 1 above)
Comment (Ruben-VandeVelde) [reply Mathlib/Algebra/Algebra/Equiv.lean:199]: You need `_root_.` here or `protected` on the deprecated lemma - either is enough to make lean pick up the correct declaration.

**Review by Ruben-VandeVelde — COMMENTED** [review]
maintainer merge

File: Mathlib/Algebra/Algebra/Equiv.lean [hunk 2]
```diff
@@ -218,8 +222,9 @@ theorem commutes : ∀ r : R, e (algebraMap R A₁ r) = algebraMap R A₂ r :=
 #align alg_equiv.commutes AlgEquiv.commutes
 
 -- @[simp] -- Porting note (#10618): simp can prove this
-theorem map_smul (r : R) (x : A₁) : e (r • x) = r • e x := by
-  simp only [Algebra.smul_def, map_mul, commutes]
+@[deprecated _root_.map_smul (since := "2024-06-20")]
+theorem map_smul (r : R) (x : A₁) : e (r • x) = r • e x :=
```
Comment (Vierkantor) [inline Mathlib/Algebra/Algebra/Equiv.lean:226]: What do you think of making this `protected` at the same time? Will save us from removing a few `_root_` qualifiers.

```suggestion
protected theorem map_smul (r : R) (x : A₁) : e (r • x) = r • e x :=
```

File: Mathlib/Algebra/Algebra/Equiv.lean [hunk 3]
```diff
@@ -228,9 +233,10 @@ nonrec theorem map_sum {ι : Type*} (f : ι → A₁) (s : Finset ι) :
   map_sum e f s
 #align alg_equiv.map_sum AlgEquiv.map_sum
 
+@[deprecated _root_.map_finsupp_sum (since := "2024-06-20")]
 theorem map_finsupp_sum {α : Type*} [Zero α] {ι : Type*} (f : ι →₀ α) (g : ι → α → A₁) :
```
Comment (Vierkantor) [inline Mathlib/Algebra/Algebra/Equiv.lean:237]: idem:
```suggestion
protected theorem map_finsupp_sum {α : Type*} [Zero α] {ι : Type*} (f : ι →₀ α) (g : ι → α → A₁) :
```

**Review by Vierkantor — APPROVED** [review]
LGTM otherwise, thanks!

bors d+

**Comment by Ruben-VandeVelde** [comment]
bors r+

There's also `Matrix.map_mul` that I'm not going to touch right now

**Pull request closed without merging**